pkg/controller/inferenceservice/resources/knative/service.go

```diff
@@ -310,14 +310,17 @@ func (c *ServiceBuilder) CreateExplainerService(name string, metadata metav1.Obj
 	return service, nil
 }
 
-func (c *ServiceBuilder) buildAnnotations(metadata metav1.ObjectMeta, minReplicas int, maxReplicas int) (map[string]string, error) {
+func (c *ServiceBuilder) buildAnnotations(metadata metav1.ObjectMeta, minReplicas *int, maxReplicas int) (map[string]string, error) {
 	annotations := utils.Filter(metadata.Annotations, func(key string) bool {
 		return !utils.Includes(serviceAnnotationDisallowedList, key)
 	})
 
-	if minReplicas != 0 {
-		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(minReplicas)
+	if minReplicas == nil {
+		annotations[autoscaling.MinScaleAnnotationKey] = fmt.Sprint(constants.DefaultMinReplicas)
+	} else if *minReplicas != 0 {
```

Comment:
Why not set it to 0 if `minReplicas` is defined as 0?

Reply:
I believe this can cause an issue when `minReplicas` was first set to `1` on an `InferenceService` and then modified to 0 to enable scale-to-zero. Since the `minScale` annotation is not reset to `0` on the KSVC, the `minScale` will remain at `1`.